toolchain/sem_ir/inst.h

```diff
@@ -104,6 +106,16 @@ struct InstLikeTypeInfo<InstCat> : InstLikeTypeInfoBase<InstCat> {
 template <typename T>
 concept InstLikeType = requires { sizeof(InstLikeTypeInfo<T>); };
 
+// Supports storage of an `Inst` object's `kind` separate from its other
+// members.
+struct InstTypeAndArgs {
+  explicit InstTypeAndArgs(Inst inst);
+
+  TypeId type_id;
+  int32_t arg0;
```

Comment:
aside: maybe these and the fields in Inst could be `AnyRawId` now, I didn't see them when I introduced it for the inst category types.

Reply:
I think the contract may differ. `AnyRawId` seems intended to represent an `Id` type, but we allow things such as  `BoolValue` on `BoolLiteral`.